include/oneapi/dpl/internal/scan_by_segment_impl.h

```diff
@@ -371,7 +373,7 @@ struct __sycl_scan_by_segment_impl
 template <typename _BackendTag, typename Policy, typename InputIterator1, typename InputIterator2,
           typename OutputIterator, typename T, typename BinaryPredicate, typename Operator, typename Inclusive>
 OutputIterator
-__scan_by_segment_impl_common(__internal::__hetero_tag<_BackendTag>, Policy&& policy, InputIterator1 first1,
+__scan_by_segment_impl_common(__internal::__hetero_tag<_BackendTag>, const Policy& policy, InputIterator1 first1,
```

Comment:
Why do we need to change this signature as well?  A const lvalue ref can accept whatever policy type is forwarded along from this function.

Reply:
I changed this signature because the initial idea was to change signature of submitters and on one level up (their callers).
This `__scan_by_segment_impl_common` function calls some submitter `__sycl_scan_by_segment_impl` :
```C++
    __sycl_scan_by_segment_impl<Inclusive::value>()(_BackendTag{}, policy, key_buf.all_view(), value_buf.all_view(),
                                                    value_output_buf.all_view(), binary_pred, binary_op, init,
                                                    identity);
```

Reply:
Please see my comment at https://github.com/uxlfoundation/oneDPL/pull/2093#issuecomment-2700963832